### ISSUE TITLE
If an m.room.encryption event is redacted, we think the room is no longer encrypted

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#3498]

**Comment by htpl** [comment]
So, encryption appears disabled in direct chat between riot-web and riot-android if user of the riot-web desktop app accidentally redact the encryption event.

steps to reproduce:

1. redact the event about encryption on riot-web
2. restart riot-web 
3. send some messages

there would be notice that messages have not been sent, while encryption is being disabled on both devices

**Comment by lampholder** [comment]
Thanks for the comprehensive report :)

I was able to reproduce this. Messages sent after encryption had been 'disabled' triggered the 'Some of your messages have not been sent.' warning, but clicking 'Resend all' doesn't work and displays the following on the console:

`Error sending event Error: Room was previously configured to use encryption, but is no longer. Perhaps the homeserver is hiding the configuration event.`

Reemabling encryption does indeed restore room functionality.

**Comment by rubo77** [comment]
Would the reaction disable encryption successfully?

Or is the room completely disjunctionally broken then?

**Comment by turt2live** [comment]
Worse now, it has a chance at causing damage to the user's stored syncs, requiring a completely new initial sync to recover, and potentially leaving the room.

**Comment by RyanMelenaNoesis** [comment]
Any workaround for this issue?  I currently have a direct message chat that I'm unable to send messages to via the Riot.im Windows desktop client because of this.

**Comment by RyanMelenaNoesis** [comment]
Upon further investigation, I found that the `content` key of the `m.room.encryption` configuration was empty `{}` in the affected rooms.  Manually editing the `content` key to:

 `{
    "algorithm": "m.megolm.v1.aes-sha2"
  }`

corrected the issue.

**Comment by ShadowJonathan** [comment]
I don't believe this is actionable, because redacting state events will already cause a whole lot of other behaviour to go haywire, and element isn't able to compensate for it anyways.

I suggest this be fixed on the spec side, and not on element's side, as i believe it's a larger issue with 'redacting state events'.
